app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -33,6 +33,40 @@ cat << EOF > $SSL_CA_CERT
 <%= foreman_server_ca_cert %>
 EOF
 
+<% unless @repo.blank? -%>
+echo '#'
+echo '# Adding repository'
+echo '#'
+
+if [ x$ID = xrhel ] || [ x$ID = xfedora ] || test "${ID_LIKE#*rhel*}" != "$ID_LIKE" ; then
+  cat << EOF > /etc/yum.repos.d/foreman_registration.repo
+[foreman_register]
+name=foreman_register
+baseurl=<%= @repo %>
+enabled=1
+<% unless @repo_gpg_key.blank? -%>
+gpgcheck=1
+gpgkey=<%= @repo_gpg_key %>
+<% end -%>
```

Comment:
I think the default of `gpgcheck` is `1` (https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/sec-configuring_yum_and_yum_repositories suggests it is and while it's EL6 docs it matches my experiences on newer versions) so I think it should be
```suggestion
gpgcheck=<%= @repo_gpg_key.blank? ? 0 : 1 %>
<% unless @repo_gpg_key.blank? -%>
gpgkey=<%= @repo_gpg_key %>
<% end -%>
```

And the unless could probably also be dropped, so
```suggestion
gpgcheck=<%= @repo_gpg_key.blank? ? 0 : 1 %>
gpgkey=<%= @repo_gpg_key %>
```